plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/local/LocalCheInfrastructureProvisioner.java

```diff
@@ -89,7 +94,26 @@ public void provision(Environment envConfig, CheServicesEnvironmentImpl internal
                    .add(SystemInfo.isWindows() ? pathEscaper.escapePath(projectFolderVolume)
                                                : projectFolderVolume);
 
+        if (System.getenv(CHE_USER_ID) != null) {
+            addVolume(internalEnv, devMachineName, ETC_PASSWD_VOLUME);
+            addVolume(internalEnv, devMachineName, ETC_GROUP_VOLUME);
+        }
         // apply basic infra (e.g. agents)
         super.provision(envConfig, internalEnv);
     }
+
+	private void addVolume(CheServicesEnvironmentImpl internalEnv, String devMachineName, String volume) {
+        String path;
+        if (projectsVolumeOptions != null && projectsVolumeOptions.contains("Z")) {
+            path = format("%s:%s:ro,Z", volume, volume);
+        } else {
+            path = format("%s:%s:ro", volume, volume);
+        }
+        internalEnv.getServices()
+                .get(devMachineName)
```

Comment:
Why mapping of these files are needed for dev machines only and not all the machines?